core/domain/classifier_services_test.py

```diff
@@ -253,3 +253,50 @@ def test_update_of_classifier_training_jobs(self):
             classifier_services.get_classifier_training_job_by_id(job_id))
         self.assertEqual(classifier_training_job.exp_id, exp_id)
         self.assertEqual(classifier_training_job.status, test_status)
+
+    def test_retrieval_of_classifier_from_exploration_attributes(self):
+        """Test the get_classifier_from_exploration_attributes method."""
+
+        with self.assertRaisesRegexp(Exception, (
+            'Entity for class ClassifierDataModel with id '
+            'fake_id not found')):
+            classifier_services.get_classifier_by_id('fake_id')
+
+        exp_id = u'1'
+        state_name = 'Home'
+        classifier_id = 'classifier_id1'
+        classifier_id = classifier_models.ClassifierDataModel.create(
+            classifier_id, exp_id, 1, state_name,
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'][
+                'algorithm_id'], [], 1)
+        classifier_models.ClassifierExplorationMappingModel.create(
+            exp_id, 1, state_name, classifier_id)
+        classifier = (
+            classifier_services.get_classifier_from_exploration_attributes(
+                exp_id, 1, state_name))
+        self.assertEqual(classifier.exp_id, exp_id)
+        self.assertEqual(classifier.state_name, state_name)
+        self.assertEqual(classifier.id, classifier_id)
+
+
+    def test_creation_of_classifier_exploration_mapping(self):
+        """Test the create_classifier_exploration_mapping method."""
+
+        exp_id = u'1'
+        state_name = 'Home'
+        classifier_id = 'classifier_id1'
+        # Mapping does not exist yet.
+        classifier_services.create_classifier_exploration_mapping(
+            exp_id, 1, state_name, classifier_id)
+        classifier_exploration_mapping_model = (
```

Comment:
@prasanna08 . We don't need it in this test (we could generate the domain object with the present classifier_id). But, this is the procedure we need to follow generally when doing this. We usually have only <exp_id, exp_version, state_name> while retrieving classifiers. So, I usually retrieve the model and generate the domain object using the classifier_id from that mapping model.

But, you're right, its not required here, I'm removing it. Done.